source/partitions.c

```diff
@@ -201,6 +203,7 @@ static int s_on_partition_element(
     return AWS_OP_SUCCESS;
 
 on_error:
+    aws_partition_info_destroy(partition_info);
```

Comment:
at line 200, we may destroy the partition_info that already put into the hashtable from line 185, right?